logos/src/logos/pipeline/scheduler_interface.py

```
@@ -0,0 +1,296 @@
+            if not status.is_loaded:
+                return -20  # Cold start penalty (still available, just slow)
+            
+            # Bonus for warm, penalty for queue depth
+            return 10 - (status.queue_depth * 0.5)
+            
+        elif provider_type == 'azure':
+            try:
+                status = self._azure.get_model_status(model_id)
+                # Get deployment name from model registration
+                # (This requires model_id -> deployment mapping)
+                return 5 if status.is_loaded else None  # Azure always "loaded"
```

Ruff (0.14.7)

Consider moving this statement to an `else` block

(TRY300)